npm/util/osutil_linux/current-azure-chains.go

```diff
@@ -0,0 +1,62 @@
+//go:build !windows
+// +build !windows
+
+package ioutil
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/Azure/azure-container-networking/npm/util"
+	npmerrors "github.com/Azure/azure-container-networking/npm/util/errors"
+	"k8s.io/klog"
+	utilexec "k8s.io/utils/exec"
+)
+
+const (
+	azureChainGrepPattern   string = "Chain AZURE-NPM"
+	minAzureChainNameLength int    = len("AZURE-NPM")
+	// the minimum number of sections when "Chain NAME (1 references)" is split on spaces (" ")
+	minSpacedSectionsForChainLine int = 2
+)
+
+var errInvalidGrepResult = errors.New("unexpectedly got no lines while grepping for current Azure chains")
+
+func AllCurrentAzureChains(exec utilexec.Interface, defaultlockWaitTimeInSeconds string) (map[string]struct{}, error) {
```

Comment:
Is there a particular reason for us to introduce this file ? I think this function being in PMGR makes sense, since all the iptable related ones are together, ifanything this can be part of chain-management